mp4parse_capi/src/lib.rs

```diff
@@ -114,16 +115,30 @@ pub struct mp4parse_track_info {
 }
 
 #[repr(C)]
+#[derive(Default, Debug)]
+pub struct mp4parser_indice {
+    pub start_offset: u64,
+    pub end_offset: u64,
+    pub start_composition: u64,
+    pub end_composition: u64,
+    pub start_decode: u64,
+    pub sync: bool,
+}
+
+#[repr(C)]
 pub struct mp4parse_byte_data {
     pub length: u32,
+    // cheddar can't handle gereric type, so it needs to be multiple data types here.
     pub data: *const u8,
+    pub indices: *const mp4parser_indice,
 }
 
 impl Default for mp4parse_byte_data {
     fn default() -> Self {
         mp4parse_byte_data {
             length: 0,
             data: std::ptr::null_mut(),
+            indices: std::ptr::null_mut(),
```

Comment:
Hmm. I think these should both be `std::ptr::null()`. This works because `*mut` coerces to `*const` which the struct has in both cases.

My mistake. Should probably change them both here now that we know.